src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -143,11 +155,20 @@ private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener
 
     private static void sendNotifications(Run<?, ?> build, TaskListener listener)
             throws IOException, InterruptedException {
-        final SCMSource s = SCMSource.SourceByItem.findSource(build.getParent());
+        SCMSource s = SCMSource.SourceByItem.findSource(build.getParent());
+        String buildName = build.getFullDisplayName();
+
         if (!(s instanceof BitbucketSCMSource)) {
-            return;
+            if (sourceCache.containsKey(buildName)) {
+                s = sourceCache.get(buildName);
+            } else {
+                return;
+            }
+        } else {
+            sourceCache.put(buildName, s);
```

Comment:
Remember the last 1000 SCMSources so we can still update the status to Bitbucket.